Dependency Issue with `expanduser`

This issue is caused by dependency "expanduser".

## Problem
```
error[E0432]: unresolved import `pwd::Passwd`
  --> C:\Users\piliT\.cargo\registry\src\index.crates.io-6f17d22bba15001f\expanduser-1.2.2\src\lib.rs:10:5
   |
10 | use pwd::Passwd;
   |     ^^^^^^^^^^^ no `Passwd` in the root

For more information about this error, try `rustc --explain E0432`.
error: could not compile `expanduser` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-list v0.30.0`, intermediate artifacts can be found at `C:\Users\piliT\AppData\Local\Temp\cargo-installGbH1z4`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

## Platform
OS: windows 11 (Home) x86_64
Terminal: 
```
$PSVersionTable.PSVersion

Major  Minor  Patch  PreReleaseLabel BuildLabel
-----  -----  -----  --------------- ----------
7      4      6
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,7 +13,7 @@ categories = [ "development-tools::cargo-plugins", "command-line-utilities" ]
 anyhow = "1.0.95"
 clap = { version = "4.5.27", features = ["derive", "wrap_help"] }
 clap-cargo = "0.15.2"
-expanduser = "1.2.2"
+dirs = "6.0.0"
 indexmap = { version = "2.7.1", features = ["rayon"] }
 lazy_static = "1.5.0"
 rayon = "1.10.0"

--- a/src/bin/cargo-list.rs
+++ b/src/bin/cargo-list.rs
@@ -4,12 +4,12 @@ use clap::{
     builder::{Styles, TypedValueParser},
     Parser, ValueEnum,
 };
-use expanduser::expanduser;
+use dirs::home_dir;
 use indexmap::IndexSet;
 use rayon::prelude::*;
 use spinners::{Spinner, Spinners};
 use sprint::*;
-use std::collections::BTreeMap;
+use std::{collections::BTreeMap, path::PathBuf};
 use veg::colored::{ColoredString, Colorize, Veg};
 
 #[cfg(unix)]
@@ -222,8 +222,18 @@ fn main() -> Result<()> {
 
 fn inner(cli: &List) -> Result<()> {
     let mut sp = Spinner::new(Spinners::Line, "".into());
+
+    let cli_config_path = if cli.config == "~" {
+        home_dir().unwrap().join(&cli.config[1..])
+    } 
+    else if cli.config.starts_with("~") {
+        home_dir().unwrap().join(&cli.config[2..])
+    }
+    else {
+        PathBuf::from(&cli.config)
+    };
     let installed = Crates::from_include(
-        &expanduser(&cli.config)?,
+        &cli_config_path,
         &cli.include.iter().map(|x| x.as_str()).collect::<Vec<_>>(),
     )?;
     sp.stop();

--- a/src/bin/cargo-list.rs
+++ b/src/bin/cargo-list.rs
@@ -220,20 +220,23 @@ fn main() -> Result<()> {
     inner(&cli)
 }
 
+fn extenduser(path: &str) -> PathBuf {
+    if path == "~" {
+        home_dir().unwrap().join(&path[1..])
+    }
+    else if path.starts_with("~") {
+        home_dir().unwrap().join(&path[2..])
+    }
+    else {
+        PathBuf::from(&path)
+    }
+}
 fn inner(cli: &List) -> Result<()> {
     let mut sp = Spinner::new(Spinners::Line, "".into());
 
-    let cli_config_path = if cli.config == "~" {
-        home_dir().unwrap().join(&cli.config[1..])
-    } 
-    else if cli.config.starts_with("~") {
-        home_dir().unwrap().join(&cli.config[2..])
-    }
-    else {
-        PathBuf::from(&cli.config)
-    };
+
     let installed = Crates::from_include(
-        &cli_config_path,
+        &extenduser(&cli.config),
         &cli.include.iter().map(|x| x.as_str()).collect::<Vec<_>>(),
     )?;
     sp.stop();

--- a/tests/integration.rs
+++ b/tests/integration.rs
@@ -1,11 +1,11 @@
 use cargo_list::Crates;
-use expanduser::expanduser;
+use dirs::home_dir;
 use rayon::prelude::*;
 use std::collections::BTreeMap;
 
 #[test]
 fn crates_from() {
-    let path = expanduser("~/.cargo/.crates2.json").unwrap();
+    let path = home_dir().unwrap().join(".cargo/.crates2.json");
 
     match Crates::from(&path) {
         Ok(installed) => {
@@ -52,7 +52,7 @@ fn crates_from() {
 
 #[test]
 fn crates_from_include() {
-    let path = expanduser("~/.cargo/.crates2.json").unwrap();
+    let path = home_dir().unwrap().join(".cargo/.crates2.json");
 
     match Crates::from_include(&path, &["^t"]) {
         Ok(installed) => {

